### PR TITLE
odt2txt: Convert openDocuments to txt

### PR DESCRIPTION
**Pull request by albfan**
Add odt2txt to use it on git-for-windows

### CODE REVIEW

**Comment by mati865** [comment]
https://ci.appveyor.com/project/Alexpux/msys2-packages/build/1.0.990#L434

**Comment by albfan** [comment]
I need hints to do the 32 compilation. Any example in other PKGBUILD?

File: odt2txt/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+pkgname=odt2txt
+pkgver=0.5
+pkgrel=2
+pkgdesc="extracts the text out of OpenDocument Texts"
+url="https://github.com/dstosberg/odt2txt/"
+license=("GPL2")
+arch=('i686' 'x86_64')
+depends=('zlib-devel' 'libiconv-devel' 'mingw-w64-x86_64-libzip')
```
Comment (dscho) [inline odt2txt/PKGBUILD]: You cannot mix and match MSYS2 packages (`zlib-devel`, `libiconv-devel`) and MINGW packages (`${MINGW_PACKAGE_PREFIX}-libzip`; you do not want to depend on 64-bit libzip when building 32-bit odt2txt).

File: odt2txt/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Maintainer: Alberto Fanjul <albertofanjul@gmail.com>
+
+source=(https://downloads.sourceforge.net/sourceforge/${pkgname}/${pkgname}-${pkgver}.tgz)
+sha256sums=('b297752910a404c1435e703d5aedb4571222bd759fa316c86ad8c8bbe58c6d1b')
+depends=('perl' 'unzip')
+
+package() {
+  cd ${srcdir}/${pkgname}-${pkgver}
+  make BINDIR=${pkgdir}/usr/bin installbin
```
Comment (dscho) [inline odt2txt/PKGBUILD]: Did you test this locally?

File: odt2txt/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+prepare() {
+  cd "$srcdir"/$pkgname
+  patch -p1 -i "$srcdir/0001-Compile-for-Msys2.patch"
+}
+build() {
+  cd "$srcdir"/$pkgname
+  make ARCH=$arch
+}
+
+package() {
```
Comment (dscho) [inline odt2txt/PKGBUILD]: Here is a second `package`...

**Comment by dscho** [comment]
It would be much more preferably to build this as a MINGW package (i.e. without depending on the MSYS2 runtime to emulate POSIX semantics). Does odt2txt require any POSIX functions?

File: odt2txt/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# Maintainer: Alberto Fanjul <albertofanjul@gmail.com>
+
+source=(https://downloads.sourceforge.net/sourceforge/${pkgname}/${pkgname}-${pkgver}.tgz)
+sha256sums=('b297752910a404c1435e703d5aedb4571222bd759fa316c86ad8c8bbe58c6d1b')
+depends=('perl' 'unzip')
```
Comment (dscho) [inline odt2txt/PKGBUILD]: Why does it depend on Perl and unzip?

File: odt2txt/0001-Compile-for-Msys2.patch [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+@@ -73,6 +73,11 @@ ifeq ($(UNAME_O),Cygwin)
+ 	LIBS += -liconv
+ 	EXT = .exe
+ endif
++ifeq ($(UNAME_O),Msys)
++	CFLAGS += -I/mingw$(ARCH)/include -I/mingw$(ARCH)/lib/libzip/include
++	LIBS += -liconv -llibzip -L/mingw$(ARCH)/lib -lzip -lz -L/mingw$(ARCH)/lib -L/mingw$(ARCH)/bin
++	EXT = .exe
++endif
+ ifneq ($(MINGW32),)
```
Comment (dscho) [inline odt2txt/0001-Compile-for-Msys2.patch]: This, existing, clause would suggest that you *can* build odt2txt as a MINGW package, i.e. as a pure Win32 application without leaning on POSIX semantics emulated by the MSYS2 runtime. Let's build a MINGW package, then?

File: odt2txt/PKGBUILD (hunk 4 above)
Comment (albfan) [inline odt2txt/PKGBUILD]: I check again original PKGBUILD and it do not have that dependency

https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/odt2txt

Will remove it

File: odt2txt/0001-Compile-for-Msys2.patch (hunk 5 above)
Comment (albfan) [inline odt2txt/0001-Compile-for-Msys2.patch]: Yes. better that way

Let me close this and rework on mingw repo

File: odt2txt/PKGBUILD (hunk 3 above)
Comment (albfan) [inline odt2txt/PKGBUILD]: opps

File: odt2txt/PKGBUILD (hunk 1 above)
Comment (albfan) [inline odt2txt/PKGBUILD]: ok

**Comment by albfan** [comment]
Ok, now this PR is passing CI. Now I can move it to mingw repo. Will keep this until that is working and passing CI

**Comment by albfan** [comment]
hmm, Does https://github.com/Alexpux/MINGW-packages/pull/2754#issuecomment-321077648 mean it cannot be a mingw package?

**Comment by albfan** [comment]
Ok this is the right repo for this PR.

Should I add that `-lpcreposix` link?

**Comment by albfan** [comment]
@dscho There's a strange error on appveyor, https://ci.appveyor.com/project/Alexpux/mingw-packages/build/1.0.3424#L486 but seems that  Alexpux/MINGW-packages#2754 is the correct place?

**Comment by albfan** [comment]
Finally it fits as MinGW package

**Pull request closed without merging**